### ISSUE TITLE
One question about the measurement of the latency

### ISSUE THREAD

**Issue by guanchuwang** [flexflow/FlexFlow#1099]
> Here are the two tests that I am running. vLLM I am getting 5.8s to complete, and I am getting 26s with FlexFlow. I am testing on a single GPU for simplicity's sake as of right now. Do you have discord? I am part of the Open-Orca group trying to evaluate if FlexFlow could be a good inference engine for us.
> 
> Flexflow Code:
> 
> ```
> import flexflow.serve as ff
> import datasets
> import time
> 
> ff.init(
>         num_gpus=1,
>         memory_per_gpu=22048,
>         zero_copy_memory_per_node=20000,
>         tensor_parallelism_degree=1,
>         pipeline_parallelism_degree=1
>     )
> 
> # Specify the LLM
> llm = ff.LLM("decapoda-research/llama-7b-hf")
> 
> # Specify a list of SSMs (just one in this case)
> ssms=[]
> ssm = ff.SSM("JackFram/llama-160m")
> ssms.append(ssm)
> 
> # Create the sampling configs
> generation_config = ff.GenerationConfig(
>     do_sample=False, temperature=0.9, topp=0.8, topk=1
> )
> 
> # Compile the SSMs for inference and load the weights into memory
> for ssm in ssms:
>     ssm.compile(generation_config)
> 
> # Compile the LLM for inference and load the weights into memory
> compiled = llm.compile(generation_config, ssms=ssms)
> 
> prompts = ["What's the best way to cook an egg?\n"] * 10
> start_time = time.time()
> result = llm.generate(prompts)
> print("--- %s seconds ---" % (time.time() - start_time))
> ```
> 
> vLLM Code:
> 
> ```
> from vllm import LLM, SamplingParams
> import time
> 
> prompts = ["What's the best way to cook an egg?\n"] * 10
> sampling_params = SamplingParams(temperature=0.8, top_p=0.95, max_tokens=256)
> 
> llm = LLM(model="decapoda-research/llama-7b-hf", tokenizer='hf-internal-testing/llama-tokenizer')
> start_time = time.time()
> outputs = llm.generate(prompts, sampling_params)
> print("--- %s seconds ---" % (time.time() - start_time))
> 
> # Print the outputs.
> for output in outputs:
>     prompt = output.prompt
>     generated_text = output.outputs[0].text
>     print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}")
>     ```
> ```

Dear team of flexflow:

Hi! We are trying to benchmark the latency and throughput of the speculative decoding. We found this code in another issue of this repo. Could you tell us whether it is the right way to measure the latency? Is this the way that the paper (https://arxiv.org/pdf/2305.09781.pdf) used to benchmark the latency?

Thanks in advance for your help!

**Comment by ChuanhongLi** [comment]
Hi，same question about this.  Any answer?

**Comment by jiazhihao** [comment]
Hi, the issue is that FlexFlow Serve uses a batch size of 1 by default. So the 10 requests are executed sequentially, each of which takes 2.6s. If you grep `latency` in the output log, you should be able to see similar measurements.

We are currently fixing an internal bug in #1096 , which will allow FlexFlow Serve to process multiple requests in parallel.

**Comment by ChuanhongLi** [comment]
Thanks for your help！I use similar code to test the latency（GPU: RTX-4090）, but i get a confused result：
1）LLaMA-7B + 1个SSM(llama-160M), latency is about 25.1 s
2）LLaMA-7B（without ssms）, latency is about 24.8 s

speculative inference does not work well?  

The code:
import flexflow.serve as ff
import time

ff.init(
    num_gpus=1,
    memory_per_gpu=22000,
    zero_copy_memory_per_node=30000,
    tensor_parallelism_degree=1,
    pipeline_parallelism_degree=1
)

# Specify the LLM， /data/lich/llama-7b-hf
llm = ff.LLM("/data/lich/llama-7b-hf")

# Specify a list of SSMs，/data/lich/llama-160m（I have downloaded the model from https://huggingface.co/JackFram/llama-160m, and put it to the path /data/lich/）
ssms = []

# test without ssms
# ssm = ff.SSM("/data/lich/llama-160m")
# ssms.append(ssm)

# Create the sampling configs
generation_config = ff.GenerationConfig(
    do_sample=False, temperature=0.9, topp=0.8, topk=1
)

# Compile the SSMs for inference and load the weights into memory
for ssm in ssms:
    ssm.compile(generation_config)

# Compile the LLM for inference and load the weights into memory
llm.compile(generation_config, ssms=ssms)

# test data comes from WebQA
prompts = [
    "what is the name of justin bieber brother?",
    "what character did natalie portman play in star wars?",
    "what state does selena gomez?",
    "what country is the grand bahama island in?",
    "what kind of money to take to bahamas?",
    "what character did john noble play in lord of the rings?",
    "who does joakim noah play for?",
    "where are the nfl redskins from?",
    "where did saki live?"
]

start_time = time.time()
result = llm.generate(prompts)
print("--- %s seconds ---" % (time.time() - start_time))

And, the logs are as follows:
[with ssms.txt](https://github.com/flexflow/FlexFlow/files/12564493/with.ssms.txt)
[without small model.txt](https://github.com/flexflow/FlexFlow/files/12564494/without.small.model.txt)

By the way, how can i reproduce the result of the paper - SpecInfer: Accelerating Generative Large Language Model Serving with Speculative Inference and Token Tree Verification？ Using above code is OK or something else? 

Thanks in advance for your help!
